zerver/lib/cache.py

```diff
@@ -829,3 +833,15 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
         return user_function(*new_args, **new_kwargs)
 
     return wrapper
+
+
+def deferred_hook(
```

Comment:
This probably deserves a docstring in addition to the `docs/` documentation on caching.